lib/stages/VisSharedMemWriter.cpp

```diff
@@ -221,8 +240,8 @@ void VisSharedMemWriter::write_to_memory(const visFrameView& frame, uint32_t tim
     uint64_t fpga_seq = frame.metadata()->fpga_seq_start;
 
 
-    if (sem_wait(sem) == -1) {
-        FATAL_ERROR("Failed to acquire semaphore {}", _sem_name);
+    if (wait_for_semaphore() == false) {
+        FATAL_ERROR("Failed to acquire semaphore {} before metadata was updated.\n", _sem_name);
```

Comment:
FATAL_ERROR exits kotekan. This way we would notice for sure that there's a problem with the shared memory. It should be in line with the error handling above, though. I'd vote for `ERROR` log and a prometheus metric.

Reply:
I made that a FATAL_ERROR because the ring buffer got modified, but the metadata did not get updated. That seemed pretty bad. However, worst case scenario in that outcome is that the calibration broker never finds out that that data is available.